crates/re_viewer/src/ui/welcome_screen/welcome_page.rs

```diff
@@ -60,114 +42,183 @@ pub(super) fn welcome_page_ui(
                 }
             }
         });
-    });
+
+        show_example
+    })
+    .inner
+}
+
+struct WelcomePagePanel<'a> {
+    title: &'static str,
+    body: &'static str,
+    image: &'static re_ui::Icon,
+    add_buttons: Box<dyn Fn(&mut egui::Ui) -> bool + 'a>, // returns true if example must be shown
```

Comment:
Using a `bool` like this everywhere is a bit obscure, and required a bunch of comments.

An alternative is to pass around something like:

```
#[must_use]
struct WelcomeScreenResponse {
    pub go_to_example_page: bool
}
```

Reply:
Good call. I've been lazy with this one.